ext/http/00_serve.js

```diff
@@ -0,0 +1,466 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+const core = globalThis.Deno.core;
+const primordials = globalThis.__bootstrap.primordials;
+
+const { BadResourcePrototype } = core;
+import { InnerBody } from "ext:deno_fetch/22_body.js";
+import { Event } from "ext:deno_web/02_event.js";
+import { toInnerResponse } from "ext:deno_fetch/23_response.js";
+import { fromInnerRequest } from "ext:deno_fetch/23_request.js";
+import { AbortController } from "ext:deno_web/03_abort_signal.js";
+import {
+  _eventLoop,
+  _idleTimeoutDuration,
+  _idleTimeoutTimeout,
+  _protocol,
+  _readyState,
+  _rid,
+  _role,
+  _server,
+  _serverHandleIdleTimeout,
+  SERVER,
+  WebSocket,
+} from "ext:deno_websocket/01_websocket.js";
+import {
+  getReadableStreamResourceBacking,
+  readableStreamForRid,
+  ReadableStreamPrototype,
+  writableStreamForRid,
+} from "ext:deno_web/06_streams.js";
+const {
+  ObjectPrototypeIsPrototypeOf,
+  SafeSet,
+  SafeSetIterator,
+  SetPrototypeAdd,
+  SetPrototypeDelete,
+  Symbol,
+  TypeError,
+  Uint8ArrayPrototype,
+} = primordials;
+
+// Symbol we use to communicate to the rest of Deno that we handle upgrades ourselves
+const _wantsUpgrade = Symbol("_wantsUpgrade");
+const _upgraded = Symbol("_upgraded");
+
+class RequestHeaders {
+  #slabId;
+
+  get(header) {
+    const value = core.ops.op_get_request_header(this.#slabId, header);
+    return value;
+  }
+}
```

Comment:
I don't understand what this is for. Can you add a comment.

Also there is no check that once a `RequestHeaders` is created, if the `slabId` becomes invalid, this object is not notified of this.

Also add a comment on what this value represents in the case there are multiple headers with the given name. Is it the first header, the last header, or the combination of headers delimited by `, `?

Reply:
This code is actually unused right now and I neglected to remove it in favour of the existing `headerList` (which easier to handle with the existing request code). I'll get rid of it.